core/src/test/scala/org/apache/spark/scheduler/TaskSetManagerSuite.scala

```diff
@@ -809,32 +821,65 @@ class TaskSetManagerSuite extends SparkFunSuite with LocalSparkContext with Logg
   test("Kill other task attempts when one attempt belonging to the same task succeeds") {
     sc = new SparkContext("local", "test")
     sched = new FakeTaskScheduler(sc, ("exec1", "host1"), ("exec2", "host2"))
-    val taskSet = FakeTask.createTaskSet(4)
+    val taskSet = FakeTask.createTaskSet(5)
     // Set the speculation multiplier to be 0 so speculative tasks are launched immediately
     sc.conf.set("spark.speculation.multiplier", "0.0")
+    sc.conf.set("spark.speculation.quantile", "0.6")
     val clock = new ManualClock()
     val manager = new TaskSetManager(sched, taskSet, MAX_TASK_FAILURES, clock)
     val accumUpdatesByTask: Array[Seq[AccumulatorV2[_, _]]] = taskSet.tasks.map { task =>
       task.metrics.internalAccums
     }
     // Offer resources for 4 tasks to start
+    val tasks = new ArrayBuffer[TaskDescription]()
     for ((k, v) <- List(
         "exec1" -> "host1",
         "exec1" -> "host1",
+        "exec1" -> "host1",
         "exec2" -> "host2",
         "exec2" -> "host2")) {
       val taskOption = manager.resourceOffer(k, v, NO_PREF)
       assert(taskOption.isDefined)
       val task = taskOption.get
       assert(task.executorId === k)
+      tasks += task
     }
-    assert(sched.startedTasks.toSet === Set(0, 1, 2, 3))
-    // Complete the 3 tasks and leave 1 task in running
+    assert(sched.startedTasks.toSet === (0 until 5).toSet)
+    // Complete 3 tasks and leave 2 task in running
     for (id <- Set(0, 1, 2)) {
       manager.handleSuccessfulTask(id, createTaskResult(id, accumUpdatesByTask(id)))
       assert(sched.endedTasks(id) === Success)
     }
 
+    def runningTaskForIndex(index: Int): TaskDescription = {
+      val t = tasks.find { task => task.index == index && !sched.endedTasks.contains(task.taskId) }
+      t match {
+        case Some(x) => x
+        case None =>
+          throw new RuntimeException(s"couldn't find index $index in " +
+            s"tasks: ${tasks.map{t => t.index -> t.taskId}} with endedTasks:" +
+            s" ${sched.endedTasks.keys}")
+      }
+    }
+
+    // have each of the running tasks fail 3 times (not enough to abort the stage)
+    (3 until 6).foreach { attempt =>
```

Comment:
I worry about another hardcoded policy here. Fine for now, perhaps something to revisit later.